RadixWallet/Features/PreAuthorizationReview/PreAuthorizationReview.swift

```diff
@@ -0,0 +1,219 @@
+	typealias Common = InteractionReview
+	typealias Expiration = DappToWalletInteractionSubintentExpiration
+
+	struct State: Sendable, Hashable {
+		let unvalidatedManifest: UnvalidatedSubintentManifest
+		let expiration: Expiration?
+		let nonce: Nonce
+		let dAppMetadata: DappMetadata.Ledger?
+
+		var preview: PreAuthorizationPreview?
+
+		var displayMode: Common.DisplayMode = .detailed
+		var sliderResetDate: Date = .now // TODO: reset when it corresponds
+		var secondsToExpiration: Int?
+
+		// Sections
+		var sections: Common.Sections.State = .init(kind: .preAuthorization)
+
+		@PresentationState
+		var destination: Destination.State? = nil
+	}
+
+	enum ViewAction: Sendable, Equatable {
+		case appeared
+		case toggleDisplayModeButtonTapped
+		case copyRawManifestButtonTapped
+	}
+
+	@CasePathable
+	enum ChildAction: Sendable, Equatable {
+		case sections(Common.Sections.Action)
+	}
+
+	enum InternalAction: Sendable, Equatable {
+		case previewLoaded(TaskResult<PreAuthorizationPreview>)
+		case updateSecondsToExpiration(Date)
+	}
+
+	enum DelegateAction: Sendable, Equatable {
+		case failed(PreAuthorizationFailure)
+	}
+
+	struct Destination: DestinationReducer {
+		@CasePathable
+		enum State: Sendable, Hashable {
+			case signing(Signing.State)
+			case rawManifestAlert(AlertState<Never>)
+		}
+
+		@CasePathable
+		enum Action: Sendable, Equatable {
+			case signing(Signing.Action)
+			case rawManifestAlert(Never)
+		}
+
+		var body: some ReducerOf<Self> {
+			Scope(state: \.signing, action: \.signing) {
+				Signing()
+			}
+		}
+	}
+
+	@Dependency(\.continuousClock) var clock
+	@Dependency(\.pasteboardClient) var pasteboardClient
+	@Dependency(\.preAuthorizationClient) var preAuthorizationClient
+	@Dependency(\.errorQueue) var errorQueue
+
+	var body: some ReducerOf<Self> {
+		Scope(state: \.sections, action: \.child.sections) {
+			Common.Sections()
+		}
+		Reduce(core)
+			.ifLet(destinationPath, action: \.destination) {
+				Destination()
+			}
+	}
+
+	private let destinationPath: WritableKeyPath<State, PresentationState<Destination.State>> = \.$destination
+
+	func reduce(into state: inout State, viewAction: ViewAction) -> Effect<Action> {
+		switch viewAction {
+		case .appeared:
+			return .run { [state = state] send in
+				let preview = await TaskResult {
+					try await preAuthorizationClient.getPreview(.init(
+						unvalidatedManifest: state.unvalidatedManifest,
+						nonce: state.nonce
+					))
+				}
+				await send(.internal(.previewLoaded(preview)))
+			}
+
+		case .toggleDisplayModeButtonTapped:
+			switch state.displayMode {
+			case .detailed:
+				return showRawManifest(&state)
+			case .raw:
+				state.displayMode = .detailed
+				return .none
+			}
+
+		case .copyRawManifestButtonTapped:
+			guard let manifest = state.displayMode.rawManifest else {
+				assertionFailure("Copy raw manifest button should only be visible in raw mode")
+				return .none
+			}
+			pasteboardClient.copyString(manifest)
```

Comment:
This could potentially be part of RawManifestView itself, we either upgrade it to have a reducer, or we could call the client straight from the view.

Reply:
The thing is, for `TransactionReview`, we still have the button placed outside the `RawManifestView`, meaning the button tapped action is handled on `TransactionReview`. We could still delegate the call to its child `RawManifestView`. 

I already do something like that for Sections, but I mostly prefer staying away of such pattern, that's why I preferred it this way. Wdyt?

Reply:
Isn't the `Copy` button always inside the RawManifestView?

Reply:
Sorry, I had totally misunderstood your comment. Updated

Reply:
no worries :).